site/docs/plugins/entity-parser.md

```diff
@@ -0,0 +1,237 @@
+| `mention`               | `CommonEntity`      | `<a class="tg-mention" href="https://t.me/${username}"> ... </a>`                                                                                                                  |
+| `phone_number`          | `CommonEntity`      | `<a class="tg-phone-number" href="tel:${options.text}"> ... </a>`                                                                                                                  |
+| `pre`                   | `PreEntity`         | `<pre class="tg-pre-code"><code class="language-${options.entity.language}"> ... </code></pre>` or `<pre class="tg-pre"> ... </pre>`                                               |
+| `spoiler`               | `CommonEntity`      | `<span class="tg-spoiler"> ... </span>`                                                                                                                                            |
+| `strikethrough`         | `CommonEntity`      | `<del class="tg-strikethrough"> ... </del>`                                                                                                                                        |
+| `text_link`             | `TextLinkEntity`    | `<a class="tg-text-link" href="${options.entity.url}"> ... </a>`                                                                                                                   |
+| `text_mention`          | `TextMentionEntity` | `<a class="tg-text-mention" href="https://t.me/${options.entity.user.username}"> ... </a>` or `<a class="tg-text-mention" href="tg://user?id=${options.entity.user.id}"> ... </a>` |
+| `underline`             | `CommonEntity`      | `<span class="tg-bot-command"> ... </span>`                                                                                                                                        |
+| `url`                   | `CommonEntity`      | `<a class="tg-url" href="${options.text}"> ... </a>`                                                                                                                               |
+
+If you are unsure which interface is correct, refer to how the [Renderer](https://github.com/quadratz/telegram-entities-parser/blob/main/src/renderers/renderer.ts) or [RendererHtml](https://github.com/quadratz/telegram-entities-parser/blob/main/src/renderers/renderer_html.ts) is implemented.
+
+### Customize the Text Sanitizer
+
+The output text is sanitized by default to ensure proper HTML rendering and prevent XSS vulnerabilities.
+
+| Input | Output   |
+| ----- | -------- |
+| `&`   | `&amp;`  |
+| `<`   | `&lt;`   |
+| `>`   | `&gt;`   |
+| `"`   | `&quot;` |
+| `'`   | `&#x27;` |
+
+For example, the result `<b>Bold</b> & <i>Italic</i>` will be sanitized to `<b>Bold</b> &amp; <i>Italic</i>`.
```

Comment:
I'm actually dumb, but won't it be sanitized to `&lt;b&gt;Bold&lt;/b&gt; &amp; &lt;i&gt;Italic&lt;/i&gt;`
> 
```ts
"&lt;b&gt;Bold&lt;/b&gt; &amp; &lt;i&gt;Italic&lt;/i&gt;".replaceAll("&lt;", "<").replaceAll("&gt;", ">").replaceAll("&amp;", "&")
// `<b>Bold</b> &amp; <i>Italic</i>`
```

Reply:
This is correct. It only sanitizes the input text, not the result itself. For instance, if the entities are
```json
  "text": "Bold & Italic <script>",
  "entities": [
   {
    "offset": 0,
    "length": 4,
    "type": "bold"
   },
   {
    "offset": 7,
    "length": 6,
    "type": "italic"
   }
  ]
```

The original output will be:
```html
<b>Bold</b> & <i>Italic<i/> <script>
```

Then it will be sanitized into:
```html
<b>Bold</b> &amp; <i>Italic<i/> &lt;script&gt;
```